src/library/trackset/playlistfeature.cpp

```diff
@@ -270,7 +270,10 @@ void PlaylistFeature::decorateChild(TreeItem* item, int playlistId) {
 
 void PlaylistFeature::slotPlaylistTableChanged(int playlistId) {
     //qDebug() << "slotPlaylistTableChanged() playlistId:" << playlistId;
-    enum PlaylistDAO::HiddenType type = m_playlistDao.getHiddenType(playlistId);
+    enum PlaylistDAO::HiddenType type = PlaylistDAO::PLHT_UNKNOWN;
+    if (playlistId != kInvalidPlaylistId) { // type is known, save a query
```

Comment:
Do we have a benefit elsewhere to put this check into getHiddenType()? 
If not, is an assertion for that sensible?

Reply:
Good point, let's put in `PlaylistDAO::getHiddenType`.

`kInvalidPlaylistId` occurs if multiple playlists were deleted for example, so no need to assert it's valid. Is that what you meant?